schema.d.ts

```diff
@@ -138,16 +138,30 @@ export type Declaration =
  *
  * All references are required to be publically accessible, so the canonical
  * representation of a reference is the export it's available from.
- *
- * Referrences to global symbols like `Array`, `HTMLElement`, or `Event`
- *
+ * 
+ * `package` should generally refer to an npm package name. If `package` is
+ * undefined then the reference is local to this package. If `module` is
+ * undefined the reference is local to the containing module.
+ * 
+ * References to global symbols like `Array`, `HTMLElement`, or `Event` should
+ * use a `package` name of `"global:"`.
  */
 export interface Reference {
   name: string;
   package?: string;
   module?: string;
 }
 
+/**
+ * A reference to the source of a declaration or member.
+ */
+export interface SourceReference {
```

Comment:
How would a developer document a `SourceReference`?

Reply:
What do you mean "document"? A SourceReference would be like:

```json
{
  "source": {
    "href": "https://github.com/lit/lit/blob/main/packages/lit-html/src/lit-html.ts#L420"
  }
}
```

Reply:
So this would only be documented manually, right? Unless an analyzer tool creates a manifest by scraping a github directly or something

Reply:
Either, yeah. A tool could do this for sure. Our API docs have "view source" links - we could use the same techniques used to generate those to put source references in manifests.